tests/subsys/dsp/basicmath/src/q31.c

```diff
@@ -302,9 +302,9 @@ static void test_arm_shift_q31(
 	free(output);
 }
 
-DEFINE_TEST_VARIANT3(basic_math_q31, arm_shift_q31, rand, in_rand, ref_shift, 9);
-DEFINE_TEST_VARIANT3(basic_math_q31, arm_shift_q31, possat, in_maxpos, ref_shift_possat, 9);
-DEFINE_TEST_VARIANT3(basic_math_q31, arm_shift_q31, negsat, in_maxneg, ref_shift_negsat, 9);
+DEFINE_TEST_VARIANT3(basic_math_q31, zdsp_shift_q31, rand, in_rand, ref_shift, 9);
+DEFINE_TEST_VARIANT3(basic_math_q31, zdsp_shift_q31, possat, in_maxpos, ref_shift_possat, 9);
+DEFINE_TEST_VARIANT3(basic_math_q31, zdsp_shift_q31, negsat, in_maxneg, ref_shift_negsat, 9);
 
 static void test_arm_and_u32(
 	const uint32_t *input1, const uint32_t *input2, const uint32_t *ref, size_t length)
```

Comment:
I see that this file still contains some `arm_` prefixed function (i.e. CMSIS-DSP function) tests; e.g. `test_arm_and_u32`.

This is very confusing and does not seem very logical -- since this is a **ZDSP** test, it should not contain any CMSIS-DSP tests.

Since splitting this test into CMSIS-DSP and ZDSP tests would be a lot of unnecessary work with little gain, why not just add `zdsp_and_u32` and its friends to the ZDSP API? After all, these integer DSP functions can be valuable since many SIMD instructions implement integer operations.

Reply:
Thanks, I missed those in my regex for swapping the tests over.